knative-operator/pkg/common/telemetry/telemetry_ucontroller.go

```diff
@@ -0,0 +1,100 @@
+package telemetry
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/runtime"
+	kafkasourcev1beta1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1beta1"
+	eventingsourcesv1beta1 "knative.dev/eventing/pkg/apis/sources/v1beta1"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var (
+	ServingObjects = []runtime.Object{
+		&servingv1.Service{},
+		&servingv1.Revision{},
+		&servingv1.Route{},
+	}
+
+	EventingObjects = []runtime.Object{
+		&eventingsourcesv1beta1.PingSource{},
+		&eventingsourcesv1beta1.ApiServerSource{},
+		&eventingsourcesv1beta1.SinkBinding{},
+	}
+
+	KnativeKafkaObjects = []runtime.Object{
+		&kafkasourcev1beta1.KafkaSource{},
+	}
+)
+
+// newTelemetryController creates an unmanaged controller for watching Telemetry resources
+func newTelemetryController(name string, objects []runtime.Object, mgr manager.Manager, telemetry *Telemetry) (*controller.Controller, error) {
+	// Create a new controller
+	c, err := controller.NewUnmanaged(fmt.Sprintf("telemetry-resources-%s-controller-%s", name, uuid.New().String()), mgr, controller.Options{
```

Comment:
What do we gain by adding the uuid in here?

Reply:
Different names in logs.

Reply:
Do we do that for our other controllers? I'd actually not want to do that as the strings should actually be stable to allow for easy filtering across pods.